src/NuGet.Clients/NuGet.VisualStudio.Common/SolutionEventsListener.cs

```diff
@@ -14,22 +13,16 @@ namespace NuGet.VisualStudio
     /// </summary>
     public abstract class SolutionEventsListener : IVsSolutionEvents, IVsSolutionLoadEvents
     {
-        private IVsSolution _vsSolution;
+        protected IVsSolution VsSolution;
         private uint _cookie;
 
-        protected void Advise(IServiceProvider serviceProvider)
+        protected void Advise()
```

Comment:
Consider passing `IVsSolution` as an argument here. In that case you won't need to have protected data member.